### DISCUSSION TITLE
Unable to perform a backup - http: invalid Host header

### DISCUSSION THREAD

**Discussion by eliluong** [teslamate-org/teslamate#3391] [Q&A] [answered]
Hello, I am running the command, `docker-compose exec -T database pg_dump -U teslamate teslamate > teslamate_20231102.bck` to back up my database, but it returns an error, `http: invalid Host header`. Has anyone run into this error? I could not find anything online that would help resolve the issue. Thank you. 

update >> Solved by running `docker exec -it ad33e439cea1 pg_dump -U teslamate teslamate > ~/Downloads/teslamate_20231102.bck` instead.

**Comment by zaphod0** [comment]
Hi, having the exact same issue when I try to backup. I have tried runnig the command up posted in the update but no sure on the container ID. My container ID is 64 charactors long. Is ad33e439cea1 your container ID? Or how did you get this ID?

**Reply by eliluong** [reply]
Make sure Teslamate is running, then run `docker ps` and it will show all the active docker containers. Get the one for the postgres container.

```
?@?:~/Downloads/teslamate$ docker ps
CONTAINER ID   IMAGE                           COMMAND                  CREATED              STATUS                PORTS                                                                                            NAMES
daea330a7451   teslamate/teslamate:latest      "tini -- /bin/sh /en…"   About a minute ago   Up About a minute     0.0.0.0:4000->4000/tcp, :::4000->4000/tcp                                                        teslamate-teslamate-1
ad33e439cea1   postgres:13                     "docker-entrypoint.s…"   About a minute ago   Up About a minute     0.0.0.0:5432->5432/tcp, :::5432->5432/tcp                                                        teslamate-database-1
735bec1ee4d2   eclipse-mosquitto:1.6           "/docker-entrypoint.…"   About a minute ago   Up About a minute     0.0.0.0:1883->1883/tcp, :::1883->1883/tcp                                                        teslamate-mosquitto-1
fbb8cabfd650   teslamate/grafana:latest        "/run.sh"                About a minute ago   Up About a minute     0.0.0.0:3000->3000/tcp, :::3000->3000/tcp                                                        teslamate-grafana-1
```

**Reply by zaphod0** [reply, marked as answer]
Great thanks. That worked. Oddly, it took over 30 minutes compared to the usual 3 min to backup the 1GB database. I might look at porting to a new host as I am not sure what changed to cause the error in the first place. Thanks again.
